library/internal_model.c

```diff
@@ -129,6 +130,17 @@ IMPL_ENUM(ziti_session_type, ZITI_SESSION_TYPE_ENUM)
 
 IMPL_ENUM(ziti_protocol, ZITI_PROTOCOL_ENUM)
 
+bool ziti_service_has_permission(const ziti_service *service, ziti_session_type sessionType) {
+    if (sessionType == ziti_session_types.Dial) {
+        return (service->perm_flags & ZITI_CAN_DIAL) != 0;
+    }
+
+    if (sessionType == ziti_session_types.Bind) {
+        return (service->perm_flags & ZITI_CAN_BIND) != 0;
+    }
+
```

Comment:
maybe a log statement here indicating that this is an unexpected scenario? something like that?

Reply:
this is just a check, log message is where it is used